x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_drilldowns_services.ts

```diff
@@ -39,36 +39,28 @@ export class DashboardDrilldownsService {
     }
   }
 
-  setupDrilldowns(core: CoreSetup<StartDependencies>, plugins: SetupDependencies) {
-    const getStartServices = createStartServicesGetter<StartDependencies, unknown>(
-      core.getStartServices
-    );
-
-    const overlays = () => getStartServices().core.overlays;
-    const drilldowns = () => getStartServices().plugins.drilldowns;
-    const getSavedObjectsClient = () => getStartServices().core.savedObjects.client;
-    const getApplicationService = () => getStartServices().core.application;
+  setupDrilldowns(
+    core: CoreSetup<StartDependencies>,
+    { advancedUiActions: uiActions }: SetupDependencies
+  ) {
+    const start = createStartServicesGetter(core.getStartServices);
     const getDashboardUrlGenerator = () => {
-      const urlGenerator = getStartServices().plugins.dashboard.dashboardUrlGenerator;
+      const urlGenerator = start().plugins.dashboard.dashboardUrlGenerator;
       if (!urlGenerator)
         throw new Error('dashboardUrlGenerator is required for dashboard to dashboard drilldown');
       return urlGenerator;
     };
 
-    const getDataPluginActions = () => getStartServices().plugins.data.actions;
-
     const actionFlyoutCreateDrilldown = new FlyoutCreateDrilldownAction({ start });
     uiActions.addTriggerAction(CONTEXT_MENU_TRIGGER, actionFlyoutCreateDrilldown);
 
     const actionFlyoutEditDrilldown = new FlyoutEditDrilldownAction({ start });
     uiActions.addTriggerAction(CONTEXT_MENU_TRIGGER, actionFlyoutEditDrilldown);
 
     const dashboardToDashboardDrilldown = new DashboardToDashboardDrilldown({
-      getSavedObjectsClient,
+      start,
```

Comment:
I think this is a move in the wrong direction.  While it's very easy to just pass down `start` it encourages tight coupling to core.  See https://www.notion.so/Tight-Coupling-in-Kibana-fcee0454319140d39aa5e83052bba901

Reply:
Had a similar comment in a PR from Vadim, we can move this discussion outside this PR. Don't let it block you.